app/models/publication.rb

```diff
@@ -140,6 +140,8 @@ def self.oa_workflow_states
   scope :with_no_oa_locations, -> { distinct(:id).left_outer_joins(:open_access_locations).where(open_access_locations: { publication_id: nil }) }
   scope :activity_insight_oa_publication, -> { with_no_oa_locations.joins(:activity_insight_oa_files).where.not(activity_insight_oa_files: { location: nil }) }
   scope :doi_failed_verification, -> { activity_insight_oa_publication.where('doi_verified = false') }
+  scope :doi_nil_verification, -> { activity_insight_oa_publication.where(doi_verified: nil) }
+  scope :needs_doi_verification, -> { doi_nil_verification.where.not(oa_workflow_state: 'automatic DOI verification pending').or(doi_nil_verification.where(oa_workflow_state: nil)) }
```

Comment:
@ajkiessl @Smullz622 
Do we actually need the `oa_workflow_state: nil` condition here?

A `nil` value for `doi_verified` is what tells us that the job needs to be run, and the only exception to that rule that I can think of is if `oa_workflow_state` is `"automatic DOI verification pending"`. Isn't it ok if `oa_workflow_state` is any value other than `"automatic DOI verification pending"`?

In other words, is there a problem with
```ruby
scope :needs_doi_verification, -> { activity_insight_oa_publication.where(doi_verified: nil).where.not(oa_workflow_state: 'automatic DOI verification pending') }
```
that I'm not considering?

Reply:
That's what I had at first, but it wouldn't pick up publications where `oa_workflow_state` was nil. It only pulled ones that had a string & that string was not `"automatic DOI verification pending"`. Once I added the `oa_workflow_state: nil` condition, it pulled the appropriate publications.

Reply:
Iirc this is what we were confused with yesterday.  The scope you have above wouldn't include publications with `oa_workflow_state: nil`.  Not sure why, but it wasn't working in the tests.

Reply:
ohhh right. Generating the SQL for the scope that I suggested yields
```sql
SELECT DISTINCT "publications".* FROM "publications" INNER JOIN "activity_insight_oa_files" ON "activity_insight_oa_files"."publication_id" = "publications"."id" LEFT OUTER JOIN "open_access_locations" ON "open_access_locations"."publication_id" = "publications"."id" WHERE "open_access_locations"."publication_id" IS NULL AND "activity_insight_oa_files"."location" IS NOT NULL AND "publications"."doi_verified" IS NULL AND "publications"."oa_workflow_state" != 'automatic DOI verification pending'
```
with ActiveRecord's `.not` translating to the SQL comparison operator `!=` for `oa_workflow_state` which returns `NULL` instead of `true` when a record's `oa_workflow_state` is `NULL`.

So using the ActiveRecord `.or` works, but (and this is just my opinion) I would find it easier to read and understand if it was all in one scope (since I don't think that we need the intermediate `doi_nil_verification` scope for anything else) and written in raw SQL. So then you could do the SQL `OR` in combination with the `AND`, or maybe even better - you could use a different method of comparison:
```ruby
scope :needs_doi_verification, -> { activity_insight_oa_publication.where(doi_verified: nil).where(%{oa_workflow_state IS DISTINCT FROM 'automatic DOI verification pending'}) }
```
I think that does the trick.